packages/components-core/src/components/AnchorMenu/index.js

```diff
@@ -66,44 +70,86 @@ export const AnchorMenu = ({
       const containerBottom =
         container?.getBoundingClientRect().bottom - subsHeight;
       if (containerTop < 0 && containerBottom > 0) {
-        curSection = targetIdName;
+        newState.activeContainer = targetIdName;
       }
     });
 
-    setActualContainer(curSection);
+    setState(prevState => ({
+      ...prevState,
+      ...newState,
+    }));
+  };
+
+  const throttleWindowScroll = () => {
+    const timeout = 150;
+    // prevent function from being called excessively
+    throttle(handleWindowScroll, timeout);
+    // ensure function executes after scrolling stops
+    debounce(handleWindowScroll, timeout);
   };
 
   // Set any ASU Header on the document
-  const setHeader = () => {
+  const getHasHeader = () => {
     const pageHeader =
       document.getElementById("asu-header") ||
       document.getElementById("headerContainer") ||
       document.getElementById("asuHeader");
-    setHasHeader(!!pageHeader);
+    return (!!pageHeader);
   };
 
   // Sets prop for styled-component to change anchor menu style
   // based on if it requires different spacing for the ASU Header
-  const findContainersRequiringAltMenuSpacing = () => {
+  const getHasDegreeContainer = () => {
```

Comment:
@scott-williams-az When originally developed, I asked @davidornelas11 to change this function name to something more general, since it's possible we may discover other use cases and having a general name means we're more likely to see the opportunity to leverage it rather than developing a parallel solution. Original request at https://github.com/ASU/asu-unity-stack/pull/847#pullrequestreview-1140926753

Thoughts, Scott, Dave? I'm inclined to keep it a general name as it then "feels" more loosely coupled and extendable, conceptually.

Reply:
This function is very specifically looking for an element with ID `degreeDetailPageContainer` so a specific function name I think is OK.
However...
At least now there is only one call to this function so we probably do not need a function at all.
Just do the logic in place at line 135
~hasDegreeContainer: getHasDegreeContainer(),~
`hasDegreeContainer: (!!document.getElementById("degreeDetailPageContainer")),`

Reply:
I think what you suggest makes sense in handling it as a one-off in one line. If we find ourselves repeating this in the future, then would be the time to build out something more extendable.